tests/gdb-tests/tests/heap/test_heap.py

```diff
@@ -14,53 +14,102 @@
 
 def generate_expected_malloc_chunk_output(chunks):
     expected = {}
+
+    size = int(
+        chunks["allocated"][
+            "mchunk_size"
+            if "mchunk_size" in (f.name for f in chunks["allocated"].type.fields())
+            else "size"
+        ]
+    )
+    real_size = size & (0xFFFFFFFFFFFFFFF - 0b111)
     expected["allocated"] = [
         "Allocated chunk | PREV_INUSE",
         f"Addr: {chunks['allocated'].address}",
-        f"Size: 0x{int(chunks['allocated']['mchunk_size' if 'mchunk_size' in (f.name for f in chunks['allocated'].type.fields()) else 'size']):02x}",
+        f"Size: 0x{real_size:02x} (with flag bits: 0x{size:02x})",
         "",
     ]
 
+    size = int(
+        chunks["tcache"][
+            "mchunk_size"
+            if "mchunk_size" in (f.name for f in chunks["tcache"].type.fields())
+            else "size"
+        ]
+    )
+    real_size = size & (0xFFFFFFFFFFFFFFF - 0b111)
     expected["tcache"] = [
         f"Free chunk ({'tcachebins' if pwndbg.heap.current.has_tcache else 'fastbins'}) | PREV_INUSE",
         f"Addr: {chunks['tcache'].address}",
-        f"Size: 0x{int(chunks['tcache']['mchunk_size' if 'mchunk_size' in (f.name for f in chunks['tcache'].type.fields()) else 'size']):02x}",
+        f"Size: 0x{real_size:02x} (with flag bits: 0x{size:02x})",
         f"fd: 0x{int(chunks['tcache']['fd']):02x}",
         "",
     ]
 
+    size = int(
+        chunks["fast"][
+            "mchunk_size"
+            if "mchunk_size" in (f.name for f in chunks["fast"].type.fields())
+            else "size"
```

Comment:
Lol this should really be some lambda but since it is a test code I'm okay with leaving it as it is (feel free to send a PR improving that if u want)